File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_066.cpp
TEST_F(SlidesIntegrationTest, SlideName) {
    Presentation pres;
    pres.slides()[0].set_name("MySlide");
    EXPECT_EQ(pres.slides()[0].name(), "MySlide");

    auto pres2 = save_and_reopen(pres);
    EXPECT_EQ(pres2.slides()[0].name(), "MySlide");
}

File: data/imported/knowledge/slides/python/merged/snippets/snippet_066.py
# Adapted from aspose.org: knowledge/slides/python/merged/snippets/snippet_066.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_solid_background_on_layout(tmp_pptx):

    """Solid-colour background on layout slide round-trips."""

    pres = Presentation()

    layout = pres.slides[0].layout_slide

    layout.background.type = BackgroundType.OWN_BACKGROUND

    layout.background.fill_format.fill_type = FillType.SOLID

    layout.background.fill_format.solid_fill_color.color = Color.coral



    pres2 = tmp_pptx(pres)

    bg = pres2.slides[0].layout_slide.background

    assert bg.type == BackgroundType.OWN_BACKGROUND

    assert bg.fill_format.fill_type == FillType.SOLID

    c = bg.fill_format.solid_fill_color.color

    assert c.r == 255 and c.g == 127 and c.b == 80

    pres2.dispose()